src/exclude_set.c

```diff
@@ -0,0 +1,77 @@
+#define _DEFAULT_SOURCE
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <traildb.h>
+
+#include "safeio.h"
+#include "judy_128_map.h"
+#include "exclude_set.h"
+
+
+typedef struct exclude_set_t {
+    struct judy_128_map uuids;
+} exclude_set_t;
+
+void free_exclude_set(exclude_set_t *s) {
+    if (s)
+        j128m_free(&s->uuids);
+
+    free(s);
+}
+
+exclude_set_t *parse_exclude_set(const char *path) {
+    char buf[250] = {0};
+
+    exclude_set_t tmp_res = {0};
+
+    FILE *f = fopen(path, "rb");
+    CHECK(f, "Cannot open %s", path);
+
+    j128m_init(&tmp_res.uuids);
+
+    uint64_t lineno = 1;
+    while (fgets(buf, sizeof(buf), f)) {
+        char *pbuf = buf;
+        __uint128_t uuid = 0;
+
+        CHECK(0 == tdb_uuid_raw((uint8_t *)pbuf, (uint8_t *)&uuid),
+              "invalid format on line %" PRIu64 " in exclude file %s (should be uuid)",
+              lineno, path);
+
+        if (uuid)
+            *j128m_insert(&tmp_res.uuids, uuid) = 1;
+
+        lineno++;
+    }
+
+    CHECK(lineno-1 == j128m_num_keys(&tmp_res.uuids),
+          "duplicate entries in exclude file %s (%" PRIu64 " lines containing %" PRIu64 " uuids)",
+          path, lineno-1, j128m_num_keys(&tmp_res.uuids));
+
+
+    fprintf(stderr, "read %" PRIu64 " uuids from %s\n", j128m_num_keys(&tmp_res.uuids), path);
+
+    exclude_set_t *res = calloc(sizeof(exclude_set_t), 1);
+    memcpy(res, &tmp_res, sizeof(exclude_set_t));
+    return res;
```

Comment:
This can be a malloc since we immediately overwrite the memory

Reply:
Actually why are we doing this with the tmp? Can we just allocate res upfront?

Reply:
Sounds good, copied some code from the `window_set` implementation, will look why it's done like this.

Reply:
Probably it was made like that so the struct gets automatically cleaned in case a check fails.